src/components/body/body.component.ts

```diff
@@ -103,11 +108,28 @@ export class DataTableBody implements OnInit, OnDestroy {
       this.hideIndicator();
 
       if(this.state.options.scrollbarV && action.type === 'pager-event') {
-        const offset = (this.state.options.rowHeight * action.limit) * action.offset;
-        this.scroller.setOffset(offset);
+        // First get the row Index that we need to move to.
+        const rowIndex = action.limit * action.offset;
+        // const offset = (this.state.options.rowHeight * action.limit) * action.offset;
+        this.scroller.setOffset(this.state.rowHeightsCache.query(rowIndex - 1));
       }
     });
 
+    this.sub.add(this.state.onExpandChange.subscribe( (expandedState) => {
+      // If there was more than one row expanded then there was a mass change
+      // in the data set hence adjust the scroll position.
+      if (expandedState.rows.length > 1) {
+        console.log(expandedState.currentIndex);
```

Comment:
@amcdnl @ocombe - my bad.  Not sure how it got missed :)   Thanks for pointing it out and pushing the fix.